gpt4all-chat/src/database.cpp

```diff
@@ -2341,16 +2348,19 @@ QList<int> Database::searchBM25(const QString &query, const QList<QString> &coll
 
         if (sqlQuery.next()) {
             // Save the query that was used to produce results
+            foundQuery = true;
             bm25q = bm25Query;
             break;
         }
     }
 
-    do {
-        const int chunkId = sqlQuery.value(0).toInt();
-        const float score = sqlQuery.value(1).toFloat();
-        results.append({chunkId, score});
-    } while (sqlQuery.next());
+    if (foundQuery) {
```

Comment:
Technically this boolean is redundant. We could consider directly checking whether the query is positioned on a valid record:
```suggestion
    if (sqlQuery.at() != QSql::AfterLastRow) {
```